src/core/georeferencing.cpp

```diff
@@ -884,6 +1013,8 @@ bool Georeferencing::setProjectedCRS(const QString& id, QString spec, std::vecto
 			if (ok && state != Normal)
 				setState(Normal);
 		}
+		if (!ok)
+			updateGridCompensation();
```

Comment:
Q: Why do we call updateGridCompensation() after an unsuccessfull change of the CRS, but not after a successful change?

Reply:
When the change is successful, the call to `setState(Normal)` calls updateGridCompensation, so a direct call is not needed. When the change is unsuccessful, updateGridCompensation is still needed to set convergence and grid_scale_factor consistent with the CRS, even if they are trivially 0 and 1.0 respectively.

I hope this helps. I am away from my office for a few days, so I do not have the opportunity just now to re-check all the paths through this section of code.